app/src/main/java/com/neilturner/aerialviews/providers/ImmichMediaProvider.kt

```diff
@@ -38,74 +48,92 @@ class ImmichMediaProvider(
         }
     }
 
-    private object RetrofitInstance {
-        fun getInstance(host: String): Retrofit =
-            Retrofit
-                .Builder()
-                .baseUrl(host)
-                .addConverterFactory(GsonConverterFactory.create())
-                .build()
-    }
-
     private interface ImmichService {
         @GET("/api/shared-links/me")
-        suspend fun getAlbum(
+        suspend fun getSharedAlbum(
             @Query("key") key: String,
             @Query("password") password: String?,
         ): Response<Album>
+
+        @GET("/api/albums")
+        suspend fun getAlbums(@Header("x-api-key") apiKey: String): Response<List<Album>>
+
+        @GET("/api/albums/{id}")
+        suspend fun getAlbum(
+            @Header("x-api-key") apiKey: String,
+            @Path("id") albumId: String
+        ): Response<Album>
     }
 
     private fun parsePrefs() {
         server = prefs.scheme?.toStringOrEmpty()?.lowercase() + "://" + prefs.hostName
-
-        val pattern = Regex("(^/)?(share/)?(.*)")
-        val replacement = "$3"
-        key = pattern.replace(prefs.pathName, replacement)
     }
 
     private fun getApiInterface() {
+        val trustAllCerts = arrayOf<TrustManager>(object : X509TrustManager {
```

Comment:
Is those code needed to ignore invalid or local SSL certs?

Reply:
Yes, if there is a better method I think it should be utilized instead. Without this code I was getting the following error: 

`javax.net.ssl.SSLHandshakeException: Chain validation failed
Caused by: java.security.cert.CertificateException: Chain validation failed`